src/jllvm/materialization/CodeGenerator.cpp

```diff
@@ -892,7 +910,8 @@ void CodeGenerator::generateInstruction(ByteCodeOp operation)
                 [&](OneOf<IfICmpGe, IfGe>) { predicate = llvm::CmpInst::ICMP_SGE; });
 
             llvm::Value* cond = m_builder.CreateICmp(predicate, lhs, rhs);
-            m_builder.CreateCondBr(cond, basicBlock, next);
+            m_builder.CreateCondBr(cond, target, next);
+            end = true;
```

Comment:
These explicitly fall through, so `end = true;` is incorrect here (although it leads to no bug since it'll immediately find a basic block afterwards)

Reply:
Although.... it depends on our definition of `end`. If we say `end` of basic block then its correct, while if we say "possible end of straight compiled instruction stream" then its incorrect.

I am guessing you are using the former definition which is probably more intuative on second thought. Could you maybe add comments to `generateInstructions` then and rename the variable to explain the meaning of the `bool` value?